### ISSUE TITLE
database code improvements

### ISSUE THREAD

**Issue by grooverdan** [dlatk/dlatk#21]
After answering  the following question on database setup on dlatk I'd like to offer some suggestions.
https://dba.stackexchange.com/questions/273751/dlatk-resolving-1045-access-denied-for-user-rootlocalhost-using-passwo

### Group database: host, user, password and database name '

For consistency with other application that deal with database, these 4 items should be the base minimum on configuration and command line options to connect to a database.

Defaulting to the current user is undocumented as is the empty password requirement.

### mm.standardizeTable

`ALTER TABLE %s CHARACTER SET %s COLLATE %s ENGINE=%s`. The default character set/collation for a table only affect newly created colummns in the table. They don't actually change the existing columns. See [example](https://dbfiddle.uk/?rdbms=mariadb_10.5&fiddle=1f8b17c95e009e1fd5168e54ffbf2fe5)

Remove MyISAM the default engine. Leave it blank. Different MySQL versions have different defaults so don't try to change engines.

### disable/renable keys

Rexample if needed per;

https://github.com/dlatk/dlatk/blob/f3199f483802c94ac8cc943f7b2ff6aead3d848a/dlatk/featureRefiner.py#L111

To support faster loading on transactional based tables recommend wrapping this look in [a transaction](https://dev.mysql.com/doc/connectors/en/connector-python-api-mysqlconnection-start-transaction.html).

https://github.com/dlatk/dlatk/blob/f3199f483802c94ac8cc943f7b2ff6aead3d848a/dlatk/featureRefiner.py#L115

(enable/disable keys should be on the outer side of this if still needed).

### group alter table

https://github.com/dlatk/dlatk/blob/f3199f483802c94ac8cc943f7b2ff6aead3d848a/dlatk/featureRefiner.py#L169

`ALTER TABLE` can take comma separated  list of modifications.

**Comment by izahn** [comment]
> After answering the following question on database setup on dlatk I'd like to offer some suggestions.
> https://dba.stackexchange.com/questions/273751/dlatk-resolving-1045-access-denied-for-user-rootlocalhost-using-passwo
> 
> ### Group database: host, user, password and database name '
> For consistency with other application that deal with database, these 4 items should be the base minimum on configuration and command line options to connect to a database.
> 
> Defaulting to the current user is undocumented as is the empty password requirement.

Yeah, this bit us too. The password will be picked up if you have it in your `~/.my.cnf` config, but the username will not be. That caused us a lot of grief trying to figure out why we couldn't connect to our database!

**Comment by sjgiorgi** [comment]
We've made some updates here. DLATK no longer takes command line arguments for specific MySQL settings (e.g. `--host`). We also no longer default to the current user. All settings are read from a config file, which defaults to `~/.my.cnf`. You can [override this default](http://dlatk.wwbp.org/fwinterface/fwflag_mysql_config_file.html) with `--mysql_config_file`. 

We also have SQLite capabilities for the standard DLATK pipeline (not all features are implemented). I wrote up some quick notes on all of the above [here](http://dlatk.wwbp.org/tutorials/tut_dataengine.html).
